Content.Client/GameObjects/Components/Items/HandsComponent.cs

```diff
@@ -284,7 +356,39 @@ public Hand(HandsComponent parent, SharedHand hand, IEntityManager manager, Hand
         public int Index { get; }
         public string Name { get; }
         public HandLocation Location { get; set; }
-        public IEntity? Entity { get; set; }
+
+        public IEntity? Entity
+        {
+            get => _entity;
+            set
+            {
+                if (_entity == value)
+                {
+                    return;
+                }
+
+                var old = _entity;
+                _entity = value;
+
+                var interactionSystem = EntitySystem.Get<SharedInteractionSystem>();
```

Comment:
You could move that under the `if(...){return;}`

Reply:
Done.